### PR TITLE
Modiy nix Files

### PR DESCRIPTION
**Pull request by Tanish2002**
* did a `cargo update`
* did some `cargo sqlx` magic and probably broke base.db, idk :p
* updated `shell.nix` `default.nix` and `flake.nix`
* We can now use `nix build` to build robbb with all checks(clippy,
cargofmt, etc) and build all the cargo deps separately so they can be
re-used. Reproducible builds!
* We can also use `nix run` to build just like `nix build` and run the
built package
* `nix-shell` and `nix-build` should just work like before, they
use flakes.lock now so removed the sources.nix and sources.json files.

Future possibility:
* Use nix for workflows: For any push, pr, etc it'll build and perform all checks using just one workflow
* We could use cachix to push cargo deps build cache for faster build incase deps aren't updated.
* The vm could also use nix(reproducible builds!) also if we use cachix then deployment would be faster.

### CODE REVIEW

**Comment by Tanish2002** [comment]
@elkowar review kthx

**Comment by Tanish2002** [comment]
also @fortuneteller2k can you also review my nix-fu

File: flake.nix [hunk 1]
```diff
@@ -1,15 +1,78 @@
...
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ (import rust-overlay) ];
+        };
+
+        # Common Args
+        commonArgs = {
+          pname = "robbb";
+          # Will Require `--impure` to be passed to it can read that ENV VAR
+          version = if builtins.getEnv "VERSION" != "" then
```
Comment (fortuneteller2k) [inline flake.nix:30]: might as well set it to some arbitrary version, instead of reading from env

File: flake.nix [hunk 2]
```diff
@@ -1,15 +1,78 @@
...
-    {
-      devShell = import ./shell.nix { inherit pkgs; };
-    }
-  );
+  outputs = { self, nixpkgs, utils, crane, rust-overlay, ... }:
+    utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ (import rust-overlay) ];
```
Comment (fortuneteller2k) [inline flake.nix:23]:
```suggestion
          overlays = [ rust-overlay.overlays.default ];
```

File: flake.nix [hunk 3]
```diff
@@ -1,15 +1,78 @@
...
+
+        # `nix develop`
+        devShells.default = pkgs.mkShell {
+          inputsFrom = builtins.attrValues self.checks;
+
+          shellHook = ''
+            export $(cat .env)
+          '';
+          # Extra inputs can be added here
+          inherit (commonArgs) nativeBuildInputs buildInputs;
```
Comment (fortuneteller2k) [inline flake.nix:75]:
```suggestion
          packages = commonArgs.nativeBuildInputs ++ commonArgs.buildInputs;
```

File: flake.nix [hunk 4]
```diff
@@ -1,15 +1,78 @@
...
+
+        # Common Args
+        commonArgs = {
+          pname = "robbb";
+          # Will Require `--impure` to be passed to it can read that ENV VAR
+          version = if builtins.getEnv "VERSION" != "" then
+            builtins.getEnv "VERSION"
+          else
+            "0.0.1";
+          src = ./.;
```
Comment (fortuneteller2k) [inline flake.nix:34]:
```suggestion
          src = builtins.path { path = pkgs.lib.cleanSource ./.; name = "robbb"; };
```

**Review by fortuneteller2k — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 1 above)
Comment (Tanish2002) [reply flake.nix:30]: not sure about this myself. I did set a fallback to "0.0.1"

Comment (fortuneteller2k) [reply flake.nix:30]: `self.rev` is probably the best move

Comment (Tanish2002) [reply flake.nix:30]: So just remove the impure part and use `self.rev`?

Comment (fortuneteller2k) [reply flake.nix:30]: yes

**Pull request merged**